### ISSUE TITLE
Identify the Problem

### ISSUE THREAD

**Issue by RogierFrans** [RogierFrans/DBA3803#1]
Goal: Choose a real-world problem or area where predictive analytics can be applied.

Everyone add their real-world problem and then we can decide in the next meeting what to choose. I only have some datasets now and still need to add a real-world problem.

1. KIVA small loans dataset: https://kiva.org/build
2. Inside Airbnb dataset: https://insideairbnb.com/
3. LinkedIn dataset: https://www.reddit.com/r/datasets/s/rYc4sMpqPG

Thanks!

**Comment by RogierFrans** [comment]
For the KIVA dataset we see a lot of characteristics of the different kind of loans. A problem statement is to predict how long it will take until the loan is fully funded based on the characteristics. By predicting this we can advise the user that it will take longer than usual or that it might not be worthwhile to start the loan. This can better the decision maker. 

Another option for the KIVA dataset is to predict how much one should ask based on their category and country where the asker is from. 

But I think there are more clearer predictive problems that can be though about.

**Comment by RogierFrans** [comment]
I also saw this dataset about Global Black Money Transactions: https://www.kaggle.com/datasets/waqi786/global-black-money-transactions-dataset. It has a risk score with every transaction it is done, including a legal or illegal label. I don't know if this is a real-life dataset, but by using this set we can easily make a prediction model for each transaction how high the risk is and if the money is legal or illegal.

**Comment by dziedziic** [comment]
I have found this interesting dataset about the flight delays on kaggle

https://www.kaggle.com/datasets/ulrikthygepedersen/airlines-delay/data

The problem statement would be to predict either the flight will be delayed or not (i.e using logistic regression /KNN classifier or decision tree). By providing early warnings this model can help both passengers to make informed decisions and enable the airlines to optimise resource management, reduce costs and improve overall efficiency. I found this topic quite interesting:))

**Comment by RogierFrans** [comment]
- [x] Write an introduction -> Rogier
- [x] Problem Statement and why it is important for business application. -> Jakub
- [x] Possible methods -> Finian

Deadline: 12 october

**Comment by dziedziic** [comment]
Problem Statement

The objective is to develop a model that predicts whether a flight will be delayed using key features such as flight time, airline, departure and arrival airports, day of the week, and flight duration. By analyzing historical flight data, we aim to classify flights as either delayed or on time, helping both airlines and passengers anticipate potential disruptions.

2. Why is it important?

Predicting flight delays is crucial for both airlines and passengers. For passengers, delays can be frustrating, leading to missed connections, cancellations, and unplanned disruptions. By predicting delays in advance, airlines can notify passengers, enabling them to make better travel arrangements, improving customer satisfaction. For airlines, predicting delays allows better operational efficiency. Airlines can use the insights to optimize scheduling, allocate resources, and manage staffing. In doing so, they can reduce the cascading effects of delays, such as crew shortages, gate availability, and customer compensation costs. Moreover, accurate delay predictions can help mitigate financial losses associated with fuel costs, rebooking, and customer dissatisfaction.



Should I make it longer or is it enough?

**Comment by RogierFrans** [comment]
**Introduction.**

Each year, millions of travellers face an all-too-common frustration: flight delays. In 2023 alone, approximately 24.7% of all flights at Changi Airport were delayed by at least 15 minutes (https://mot.gov.sg/news/details/written-reply-to-parliamentary-question-on-data-on-number-of-flights-in-and-out-of-changi-airport-that-were-delayed-or-cancelled), affecting thousants of passenger and costing airline millions of dollars (even more for countries in Europe, because of the Flight Compensation Regulation 261/2004). These disruptions not only cause inconvenience to passengers, such as missed connections and extended layovers, but also ripple through the airline industry, leading to operational inefficiencies, crew shortages, and increased fuel costs (Wu, C.-L. (2010). Airline Operations and Delay Management: Insights from Airline Economics, Networks and Strategic Schedule Planning (1st ed.). Routledge. https://doi-org.eur.idm.oclc.org/10.4324/9781315566467).

Research shows that reducing 30% of the delays could benefit the American aviation industry over $38.5 billion annually, not only from direct losses but also from the broader impacts on customer satisfaction and logistical challenges and tourism spend (Peterson, E. B., Neels, K., Barczi, N., & Graham, T. (2013). The Economic Cost of Airline Flight Delay. Journal of Transport Economics and Policy, 47(1), 107–121. http://www.jstor.org/stable/24396355). As the demand for air travel continues to rise back to the levels from before covid, so does the importance of addressing this issue. A solution to this problem is using data science to predict flight delays with greater accuracy. By analyzing key variables such as flight duration, airline schedules, and day-of-week patterns, predictive models can provide actionable insights, allowing airlines to minimize disruptions and improve their service quality.

In this project, we aim to develop a predictive model for flight delays, leveraging historical flight data to help both airlines and passengers anticipate potential disruptions before they happen.

**Comment by finz7** [comment]
Possible Methods:

- Explain the analytical methods and techniques you employed in your project. This should include a description of any statistical or machine learning models, algorithms, or tools used. Provide enough detail for the reader to understand your approach but avoid unnecessary technical jargon.

- One of the techniques that we have employed is mixed selection. This is done through identifying several factors that could have contributed to flight delays that has been very prevalent in Singapore. However, we know that there are a significant number of factors, small or large, that can contribute to the delay of flights in Singapore. With too many factors, this requires us to collect more data on the situation, and too many variables can also consequently lead to overfitting. However, forward and backward selection both have their own flaws. Thus, using the mixed selection approach helps to alleviate the issue, as it addresses a fundamental drawback of forward selection, which is that the terms might become insignificant after other terms have been added to the model. Mixed selection allows nonsignificant terms to be removed. 

- Another technique that we have employed in our project is to split the dataset into the training and test dataset through the 80%-20% ratio, and applying logistic regression to predict if there would be flight delays that may occur. This is important to validate if our model is successful.  After obtaining the training and test dataset, we will use logistic regression on our training dataset, and this will be used to help build a predictive model to predict the flight delays through past historical flight data from the training dataset. Through this, we can evaluate the predictive power of model on testing set. By predicting the flight delays, we can anticipate any disruptions or delays that may occur and this would aid both passengers and the airlines. 

- Finally, the last technique that we have employed in our project is the k-folds cross validation. In brief, this means that we split the training set model into k sets, and it is not as expensive as the Leave One Out cross validation method (LOOCV). Through splitting out training dataset into k sets, we can use each of these sets to validate and predict our model's performance in predicting the flight delays that may occur. We decided to split the training dataset into 5 sets as it provides a good balance between computational efficiency and model performance estimation, and also trying to alleviate the bias–variance problem such as to minimize these two errors that may hinder the algorithm from predicting beyond the training dataset. 

- For these methods, I am not sure which methods we are going to employ into our project, so I just only provided some possible methods that we can employ. In addition, as we have not employed these methods into our project yet, I just gave a brief explanation of the methods that we can use. Sorry!

**Comment by RogierFrans** [comment]
> Possible Methods:
> 
> * Explain the analytical methods and techniques you employed in your project. This should include a description of any statistical or machine learning models, algorithms, or tools used. Provide enough detail for the reader to understand your approach but avoid unnecessary technical jargon.
> * One of the techniques that we have employed is mixed selection. This is done through identifying several factors that could have contributed to flight delays that has been very prevalent in Singapore. However, we know that there are a significant number of factors, small or large, that can contribute to the delay of flights in Singapore. With too many factors, this requires us to collect more data on the situation, and too many variables can also consequently lead to overfitting. However, forward and backward selection both have their own flaws. Thus, using the mixed selection approach helps to alleviate the issue, as it addresses a fundamental drawback of forward selection, which is that the terms might become insignificant after other terms have been added to the model. Mixed selection allows nonsignificant terms to be removed.
> * Another technique that we have employed in our project is to split the dataset into the training and test dataset through the 80%-20% ratio, and applying logistic regression to predict if there would be flight delays that may occur. This is important to validate if our model is successful.  After obtaining the training and test dataset, we will use logistic regression on our training dataset, and this will be used to help build a predictive model to predict the flight delays through past historical flight data from the training dataset. Through this, we can evaluate the predictive power of model on testing set. By predicting the flight delays, we can anticipate any disruptions or delays that may occur and this would aid both passengers and the airlines.
> * Finally, the last technique that we have employed in our project is the k-folds cross validation. In brief, this means that we split the training set model into k sets, and it is not as expensive as the Leave One Out cross validation method (LOOCV). Through splitting out training dataset into k sets, we can use each of these sets to validate and predict our model's performance in predicting the flight delays that may occur. We decided to split the training dataset into 5 sets as it provides a good balance between computational efficiency and model performance estimation, and also trying to alleviate the bias–variance problem such as to minimize these two errors that may hinder the algorithm from predicting beyond the training dataset.
> * For these methods, I am not sure which methods we are going to employ into our project, so I just only provided some possible methods that we can employ. In addition, as we have not employed these methods into our project yet, I just gave a brief explanation of the methods that we can use. Sorry!

Looks good, For the method of the regression we should also explain why we use Logistic regression and nog XGBoost for example. I would suggest we say we will look at XGBoost, logistic regression and KNN model. We can then compare the results between those models.

The same applies for the selection of variables we can show in the results the extended model (model with all the sense making variables) and then a model with variables that are multicollinear or smt like that. Will explain more. 

Two step approach:

1. Delay yes or not:
1. Lasso model (logistic)
2. KNN
3. XGBoost (tree model)
4. Amount of times:
lasso (simple model)
XGBoost 
What we can do here, we can compere two models with different predictors.
